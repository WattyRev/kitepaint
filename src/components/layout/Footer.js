import React from "react";
import styled from "styled-components";
import { A, P } from "../../theme";

export const StyleWrapper = styled.div`
  text-align: center;
  border-top: 1px solid ${props => props.theme.colors.gray}
  padding: 8px;
`;

/**
 * The footer displayed at the bottom of every page.
 */
const Footer = props => (
  <StyleWrapper {...props}>
    <P>
      &copy; {new Date().getFullYear()}{" "}
      <A href="http://wattydev.com" target="_blank">
        Spencer Watson
      </A>
    </P>
  </StyleWrapper>
);

export default Footer;
