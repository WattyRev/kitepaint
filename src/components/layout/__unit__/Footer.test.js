import React from "react";
import { shallow, mount } from "enzyme";
import Theme from "../../../theme";
import Footer, { StyleWrapper } from "../Footer";

describe("Footer", () => {
  it("renders", () => {
    expect.assertions(1);
    const wrapper = shallow(<Footer className="my-footer" />);
    expect(wrapper).toMatchSnapshot();
  });
  describe("StyleWrapper", () => {
    it("renders", () => {
      expect.assertions(1);
      const wrapper = mount(<StyleWrapper theme={Theme} />);
      expect(wrapper.find("div")).toHaveLength(1);
    });
  });
});
