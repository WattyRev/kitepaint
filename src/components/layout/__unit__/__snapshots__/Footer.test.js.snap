// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    className="my-footer"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <P>
        © 
        2018
         
        <ForwardRef
          href="http://wattydev.com"
          target="_blank"
        >
          Spencer Watson
        </ForwardRef>
      </P>,
      "className": "my-footer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "© ",
          2018,
          " ",
          <ForwardRef
            href="http://wattydev.com"
            target="_blank"
          >
            Spencer Watson
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "© ",
        2018,
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Spencer Watson",
            "href": "http://wattydev.com",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "Spencer Watson",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "rules": Array [
                "
  color: ",
                [Function],
                ";
  text-decoration: none;
  margin: 0;
  cursor: pointer;

  &:hover {
    text-decoration: underline;
  }
",
              ],
            },
            "displayName": "styled.a",
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "a",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-fjdhpX",
        "isStatic": false,
        "rules": Array [
          "
  text-align: center;
  border-top: 1px solid ",
          [Function],
          "
  padding: 8px;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-fjdhpX",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <P>
          © 
          2018
           
          <ForwardRef
            href="http://wattydev.com"
            target="_blank"
          >
            Spencer Watson
          </ForwardRef>
        </P>,
        "className": "my-footer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "© ",
            2018,
            " ",
            <ForwardRef
              href="http://wattydev.com"
              target="_blank"
            >
              Spencer Watson
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "© ",
          2018,
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Spencer Watson",
              "href": "http://wattydev.com",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Spencer Watson",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "rules": Array [
                  "
  color: ",
                  [Function],
                  ";
  text-decoration: none;
  margin: 0;
  cursor: pointer;

  &:hover {
    text-decoration: underline;
  }
",
                ],
              },
              "displayName": "styled.a",
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "a",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-fjdhpX",
          "isStatic": false,
          "rules": Array [
            "
  text-align: center;
  border-top: 1px solid ",
            [Function],
            "
  padding: 8px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-fjdhpX",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
